src/shared/codelens/localLambdaRunner.ts

```diff
@@ -510,14 +516,19 @@ export const invokeLambdaFunction = async (params: {
         invoker: params.samLocalInvokeCommand,
     })
 
-    const startInvokeTime = new Date()
-    await command.execute()
+    const timelimit = params.configuration.readSetting<number>(
+        'samcli.debug.attach.timeout.millis',
+        SAM_LOCAL_PORT_CHECK_RETRY_TIMEOUT_MILLIS_DEFAULT
+    )
+    const timer = new Timeout(timelimit)
+    await command.execute(timer)
```

Comment:
these 4-5 lines are duplicated which feels like we might want some sort of utility function.

Reply:
We have two implementations of invokeLambdaFunction (for Typescript [within a class] and Python/C# [within a function, this one]) that are essentially identical. We're planning on migrating Typescript to the functional option; this is a stopgap for now.

Reply:
How far does the stop gap go? How many pieces of duplicate code will we tolerate before we do the merge?

Reply:
✔️